### PR TITLE
Change to disable region replication peer when region replication is disabled

### PR DESCRIPTION
**Pull request by bijugs**
This PR is for issue #990.

### CODE REVIEW

File: cookbooks/bcpc-hadoop/recipes/hbase_config.rb [hunk 1]
```diff
@@ -133,7 +133,7 @@
 #
 # if HBASE region replication is enabled the properties in this section will be included in hbase-site.xml
 #
-if node["bcpc"]["hadoop"]["hbase"]["region"]["replication"]["enabled"] == true then
+if node['bcpc']['hadoop']['hbase']['site_xml']['hbase.region.replica.replication.enabled'] == 'true' then
```
Comment (vt0r) [inline cookbooks/bcpc-hadoop/recipes/hbase_config.rb]: Same question about the change from the bool (see above).

File: cookbooks/bcpc-hadoop/recipes/hbase_repl.rb [hunk 2]
```diff
@@ -15,5 +16,27 @@
...
+
+#
+# To disable region replication peer
+#
+if (node['bcpc']['hadoop']['hbase']['site_xml']['hbase.region.replica.replication.enabled'] == 'false')
+  bash "disable_hbase_region_repl_peer" do
+    code <<-EOH
+      echo "disable_peer 'region_replica_replication'"|hbase shell
+      EOH
+    only_if "echo list_peers|hbase shell|grep region_replica_replication|grep ENABLED"
```
Comment (pu239ppy) [inline cookbooks/bcpc-hadoop/recipes/hbase_repl.rb]: Either way you invoke hbase shell to make a test, is there any harm to just blindly disabling without testing?

File: cookbooks/bcpc-hadoop/recipes/hbase_repl.rb [hunk 3]
```diff
@@ -15,5 +16,27 @@
...
+    user 'hbase'
+  end
+else
+  bash "enable_hbase_region_repl_peer" do
+    code <<-EOH
+      echo "enable_peer 'region_replica_replication'"|hbase shell
+      EOH
+    only_if "echo list_peers|hbase shell|grep region_replica_replication|grep DISABLED"
```
Comment (pu239ppy) [inline cookbooks/bcpc-hadoop/recipes/hbase_repl.rb]: Either way you invoke hbase shell to make a test, is there any harm to just blindly disabling without testing?

File: cookbooks/bcpc-hadoop/recipes/hbase_repl.rb (hunk 2 above)
Comment (bijugs) [reply cookbooks/bcpc-hadoop/recipes/hbase_repl.rb]: Hi @pu239ppy , with out the peer being present, ``hbase shell`` generate errors when ``disable/enable`` command is issued. The additional check is to make sure that the peer is defined and is in the correct state.

**Comment by vt0r** [comment]
After talking with @bijugs about the move of this attribute to `site_xml`, I found out we're preferring any custom attributes we add that refer to K/V pairs in XML configs will be in one place - sort of like a custom configs namespace.

***Note**: the breaking tag was added because cluster owners who have enabled region replication explicitly using the old attribute will need to change their overrides as follows:*
```ruby
node['bcpc']['hadoop']['hbase']['region']['replication']['enabled'] = true
```
becomes:
```ruby
node['bcpc']['hadoop']['hbase']['site_xml']['hbase.region.replica.replication.enabled'] = true
```

**Pull request merged**